### PR TITLE
Add php-cs-fixer & PHPCompatibility check to workflow

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- adds PHPCompatibility check
- adds PHP-CS-Fixer to workflow
  - fixes last errors, not found by `phpcs`
- updated PhpStan
  - updated to v1.9.2
  - removed experimental check and set default level to 5 (from 3 befor)
- updated complete workflow (to run a bit faster)

### Fixed Issues (if relevant)

1. Closes OpenMage/magento-lts#947

### Related PR

1. See OpenMage/magento-lts#1264
2. See OpenMage/magento-lts#1157
3. See OpenMage/magento-lts#2707

### Manual testing scenarios (*)

`php vendor/bin/php-cs-fixer fix --config=.github/php-cs-fixer-ruleset.php <path-to-files>`

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by fballiano — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php [hunk 1]
```diff
@@ -108,20 +108,19 @@ public function __construct()
         $undefinedResources = array_diff(array_keys($resources), array_keys($resourcesPermissionsMap));
 
         foreach ($undefinedResources as $undefinedResourceId) {
-            if ($this->_getFallbackHelper()->fallbackResourcePermissions(
+            # Fallback resource permissions
```
Comment (luigifab) [inline app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php:111]: `#` for comments?

File: app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php [hunk 2]
```diff
@@ -66,7 +66,8 @@ protected function _toHtml()
         $form = new Varien_Data_Form();
         foreach ([
             'from' => Mage::helper('adminhtml')->__('From'),
-            'to'   => Mage::helper('adminhtml')->__('To')] as $key => $label) {
+            'to'   => Mage::helper('adminhtml')->__('To')] as $key => $label
+        ) {
```
Comment (luigifab) [inline app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php:70]: A new line is missing after `('To')`?

File: app/code/core/Mage/Adminhtml/controllers/Newsletter/QueueController.php [hunk 3]
```diff
@@ -19,6 +19,8 @@
  * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 
+use Mage_Newsletter_Model_Queue as Queue;
```
Comment (luigifab) [inline app/code/core/Mage/Adminhtml/controllers/Newsletter/QueueController.php:22]: Why not.

File: app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php [hunk 4]
```diff
@@ -324,7 +325,8 @@ public function getUsedProducts($requiredAttributeIds = null, $product = null)
         Varien_Profiler::start('CONFIGURABLE:' . __METHOD__);
         if (!$this->getProduct($product)->hasData($this->_usedProducts)) {
             if (is_null($requiredAttributeIds)
-                and is_null($this->getProduct($product)->getData($this->_configurableAttributes))) {
+                and is_null($this->getProduct($product)->getData($this->_configurableAttributes))
```
Comment (luigifab) [inline app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php:328]: Oh no a `and` instead of '&&'. Ok this is not the goal of the PR.

**Review by fballiano — DISMISSED** [review]
LGTM

**Review by luigifab — COMMENTED** [review]
Seems good - not tested (129/226 files viewed).
For me it's strange to use `if ($a > $b && $b < $c)` without brackets: `if (($a > $b) && ($b < $c))`.

File: app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php:111]: I use it for singe linge comments ... one char less ;)

(Can you please suggest a change?)

File: app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php:70]: No. (?)

File: app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php (hunk 4 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php:328]: Good catch - for next PR :)

File: app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php (hunk 2 above)
Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php:70]: I would say:
```php
        foreach ([
            'from' => Mage::helper('adminhtml')->__('From'),
            'to'   => Mage::helper('adminhtml')->__('To')
        ] as $key => $label) {
```
Or with a variable before, but this it not important.

File: app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php (hunk 1 above)
Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php:111]: I prefer `//` but this is not important.

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php:111]: Will change in next PR.

File: app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php (hunk 4 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php:328]: Wait for next PR.

File: app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Promo/Widget/Chooser/Daterange.php:70]: This is invalid for a sniff (cant remember right now)

**Comment by sreichel** [comment]
Fixed conflicts.

**Comment by colinmollenhour** [comment]
Nice work @sreichel! 
I have not looked at every file but it looks like a good ruleset and agree in principle this is a good feature to add.

File: .github/labeler.yml [hunk 5]
```diff
@@ -603,3 +596,22 @@
 # Add Documentation Label
 'Documentation':
   - '*.md'
+
+'CI':
+  - .travis.yml
```
Comment (elidrissidev) [inline .github/labeler.yml:601]: We're using travis?

File: .github/workflows/phpstan.yml [hunk 6]
```diff
@@ -1,47 +1,24 @@
-name: "PHPStan Static Code Analyses"
+name: PHPStan Static Code Analyses
 
 on:
-  # Run on all pushes and on all pull requests.
-  # Prevent the build from running when there are only irrelevant changes.
-  push:
-    paths:
-      - '**.php'
```
Comment (elidrissidev) [inline .github/workflows/phpstan.yml:8]: Isn't this going to make them run in every PR again, even with irrelevant changes?

File: .github/labeler.yml (hunk 5 above)
Comment (sreichel) [reply .github/labeler.yml:601]: I just moved this line from top of the file,

Comment (elidrissidev) [reply .github/labeler.yml:601]: doesn't matter then, you can keep it or remove it.

File: .github/workflows/phpstan.yml (hunk 6 above)
Comment (sreichel) [reply .github/workflows/phpstan.yml:8]: Yes. It is not possible to seth path to workflows that are called from another workflow. Working on this.

File: app/code/core/Mage/Tag/Model/Resource/Tag.php [hunk 7]
```diff
@@ -85,7 +85,8 @@ protected function _beforeSave(Mage_Core_Model_Abstract $object)
             $searchTag = new Varien_Object();
             $this->loadByName($searchTag, $object->getName());
             if ($searchTag->getData($this->getIdFieldName())
-                    && $searchTag->getStatus() == $object->getPendingStatus()) {
+                    && $searchTag->getStatus() == $object->getPendingStatus()
```
Comment (kiatng) [inline app/code/core/Mage/Tag/Model/Resource/Tag.php:88]:
```suggestion
                && $searchTag->getStatus() == $object->getPendingStatus()
```

File: lib/Varien/Db/Adapter/Pdo/Mysql.php [hunk 8]
```diff
@@ -369,7 +359,7 @@ protected function _connect()
             throw new Zend_Db_Adapter_Exception('pdo_mysql extension is not installed');
         }
 
-        $hostInfo = $this->_getHostInfo(isset($this->_config['host']) ? $this->_config['host'] : (isset($this->_config['unix_socket']) ? $this->_config['unix_socket'] : null));
+        $hostInfo = $this->_getHostInfo($this->_config['host'] ?? ($this->_config['unix_socket'] ?? null));
```
Comment (kiatng) [inline lib/Varien/Db/Adapter/Pdo/Mysql.php:362]:
```suggestion
        $hostInfo = $this->_getHostInfo($this->_config['host'] ?? $this->_config['unix_socket'] ?? null);
```

File: .github/workflows/phpstan.yml (hunk 6 above)
Comment (sreichel) [reply .github/workflows/phpstan.yml:8]: Here we go ... https://github.com/sreichel/magento-lts/pull/26

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Paypal/Model/Config.php [hunk 9]
```diff
@@ -748,7 +750,8 @@ public function isMethodAvailable($methodCode = null)
                 if ($this->isMethodActive(self::METHOD_WPP_PE_DIRECT)) {
                     $result = true;
                 } elseif (!$this->isMethodActive(self::METHOD_WPP_PE_DIRECT)
-                          && !$this->isMethodActive(self::METHOD_PAYFLOWPRO)) {
+                          && !$this->isMethodActive(self::METHOD_PAYFLOWPRO)
```
[the remark this answers is not in the document]
Comment (sreichel) [reply app/code/core/Mage/Paypal/Model/Config.php]: Thanks :) :+1:

**Pull request merged**